src/mapservutil.c

```diff
@@ -89,9 +89,13 @@ void msCGIWriteError(mapservObj *mapserv) {
     return;
   }
 
-  if ((ms_error->code == MS_NOTFOUND) && (mapserv->map->web.empty)) {
-    /* msRedirect(mapserv->map->web.empty); */
-    if (msReturnURL(mapserv, mapserv->map->web.empty, BROWSE) != MS_SUCCESS) {
+  if ((ms_error->code == MS_NOTFOUND) &&
+      (mapserv->map->web.empty != NULL ||
+       CPLGetConfigOption("MS_EMPTY", NULL) != NULL)) {
+    const char *url = mapserv->map->web.empty;
```

Comment:
@rouault, the const qualifier is causing a problem with the build check since msReturnURL() is looking for a just char *. After review, I see that msReturnURL() can alter the passed url using processLine(). I'm not sure why that's necessary here but I suppose there a couple substitutions that might be useful. I guess that's why the msRedirect() call is commented out. 

With the configuration option I don't think we want that value modified so it would be necessary to duplicate the value from either the local configuration or global configuration and then free it after the redirect. I just wanted to run that change past someone as it's adding complexity.

Reply:
> I see that msReturnURL() can alter the passed url using processLine().

no, it doesn't since processLine() takes a const char* itself.

The following builds fine and should fix the issue:
```
diff --git a/src/maptemplate.c b/src/maptemplate.c
index ff2c3bc1d..7a5fb8ac3 100644
--- a/src/maptemplate.c
+++ b/src/maptemplate.c
@@ -4702,7 +4702,7 @@ int msReturnPage(mapservObj *mapserv, char *html, int mode,
   return MS_SUCCESS;
 }
 
-int msReturnURL(mapservObj *ms, char *url, int mode) {
+int msReturnURL(mapservObj *ms, const char *url, int mode) {
   char *tmpurl;
 
   if (url == NULL) {
diff --git a/src/maptemplate.h b/src/maptemplate.h
index 5eea71788..887b829c3 100644
--- a/src/maptemplate.h
+++ b/src/maptemplate.h
@@ -198,7 +198,7 @@ MS_DLL_EXPORT int checkWebScale(mapservObj *msObj);
 MS_DLL_EXPORT int setExtent(mapservObj *msObj);
 
 MS_DLL_EXPORT int msReturnPage(mapservObj *msObj, char *, int, char **);
-MS_DLL_EXPORT int msReturnURL(mapservObj *msObj, char *, int);
+MS_DLL_EXPORT int msReturnURL(mapservObj *msObj, const char *, int);
 MS_DLL_EXPORT int msReturnNestedTemplateQuery(mapservObj *msObj,
                                               char *pszMimeType,
                                               char **papszBuffer);

```